src/App.js

```diff
@@ -1,16 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
 
 const App = () => {
+  const [entries,setEntriesData] = useState(chatMessages);
+
+  const updateLikedMessagesCount = (id) => {
+    setEntriesData((prev) => {
+      return prev.map((entry) => {
+        if(id === entry.id) {
+          return {
+            ...entry,
+            liked: !entry.liked,
+          };
+        } else {
+          return entry;
+        }
+      });
+    });
+  };
+
+  const totalLikeTally = entries.reduce((total,entry) => {
+    if(entry.liked === true){
```

Comment:
Totally ok to compare  to a boolean but it would be more common to just check for truthinees:
```js
if (entry.liked) {
```